Sources/RunnerLib/VersionChecker.swift

```diff
@@ -0,0 +1,50 @@
+import DangerShellExecutor
+import Logger
+import Version
+import Foundation
+
+public enum VersionChecker {
+    public static func checkForUpdate(current currentVersionString: String) {
+        if ProcessInfo.processInfo.environment["DEBUG"] != nil {
+            return
+        }
+        guard let latestVersionString = fetchLatestVersion() else { return }
+        guard let latestVersion = Version(latestVersionString) else {
+            logger.debug("Invalid latestVersionString: (\(latestVersionString)")
+            return
+        }
+        guard let currentVersion = Version(currentVersionString) else {
+            logger.debug("Invalid currentVersionString: (\(currentVersionString)")
+            return
+        }
+        if currentVersion < latestVersion {
+            logger.logInfo("\nℹ️  A new release of danger-swift is available: \(currentVersion) -> \(latestVersion)")
+        }
+    }
+
+    static func fetchLatestVersion() -> String? {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        do {
+            let latest = try ShellExecutor().execute("curl",
+                                                     arguments: [
+                                                        "-s",
+                                                        "https://api.github.com/repos/danger/swift/releases/latest"
+                                                     ])
+                .data(using: .utf8)
+                .flatMap { try decoder.decode(Release.self, from: $0) }
+            return latest?.tagName
+        } catch {
+            logger.debug(error)
+            return nil
+        }
+    }
+}
+
+extension VersionChecker {
+    static let logger = Logger()
```

Comment:
This Logger won't have the right set up, which is made based on the parameters passed to Danger Swift, not a big deal, but would be nice to get it from the runner if possible (is created at line 25)

Reply:
https://github.com/danger/swift/pull/505#discussion_r800280200 may also resove it 😁